python/utils/submit_recovery_share.sh

```diff
@@ -46,9 +56,28 @@ if [ -z "${member_enc_privk}" ]; then
     exit 1
 fi
 
+if [ -z "${member_cert}" ]; then
+    echo "Error: No member cert in arguments (--cert)"
+    exit 1
+fi
+
+if [ -z "${member_key}" ]; then
+    echo "Error: No member key in arguments (--key)"
+    exit 1
+fi
+
+member_id=$(openssl x509 -in "${member_cert}" -noout -fingerprint -sha256 | cut -d "=" -f 2 | sed 's/://g' | awk '{print tolower($0)}')
+
 # First, retrieve the encrypted recovery share
-encrypted_share=$(curl -sS --fail -X GET "${node_rpc_address}"/gov/recovery_share "${@}" | jq -r '.encrypted_share')
+encrypted_share=$(curl -sS --fail -X GET "${node_rpc_address}"/gov/recovery_share/"${member_id}" "${@}" | jq -r '.encrypted_share')
 
 # Then, decrypt encrypted share with member private key submit decrypted recovery share
 # Note: all in one line so that the decrypted recovery share is not exposed
-echo "${encrypted_share}" | openssl base64 -d | openssl pkeyutl -inkey "${member_enc_privk}" -decrypt -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256 | openssl base64 -A | jq -R '{share: (.)}' | curl -i -sS --fail -H "Content-Type: application/json" -X POST "${node_rpc_address}"/gov/recovery_share "${@}" -d @-
\ No newline at end of file
+echo "${encrypted_share}" \
+    | openssl base64 -d \
+    | openssl pkeyutl -inkey "${member_enc_privk}" -decrypt -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256 \
+    | openssl base64 -A \
+    | jq -R '{share: (.)}' \
+    | ccf_cose_sign1 --content - --signing-cert "${member_cert}" --signing-key "${member_key}" \
```

Comment:
This implies that `ccf_cose_sign1` is in the path, e.g. via `pip install ccf`. I don't think `submit_recovery_share.sh` had such a requirement before and could be run "freely"? If so, this should be mentioned in the changelog.

Reply:
I think that's fair, we also assume jq